src/app/parcels/batch/displayComponents/BatchParcelDataGrid.tsx

```diff
@@ -37,65 +45,277 @@ export interface BatchGridDisplayRow {
+                        margin: "auto",
+                        borderColor: confirmationErrors.length == 0 ? "#a8d49c" : "#990f0f",
+                        borderWidth: "1px",
+                        borderStyle: "solid",
                     },
-                },
-                "& .MuiDataGrid-columnHeaders": {
-                    borderBottom: "3px solid",
-                    borderTop: "2px solid",
-                    borderColor: `${theme.main.border}`,
-                },
-                "& .MuiDataGrid-columnHeader": {
-                    borderLeft: "1px solid",
-                    borderRight: "1px solid",
-                    borderColor: `${theme.main.border}`,
-                },
-                "& .MuiDataGrid-columnHeaderTitle": {
-                    fontWeight: "bold",
-                },
-                "& .Mui-error": {
-                    backgroundColor: `${theme.error}`,
-                    color: `${theme.text}`,
-                },
-                border: "1px solid",
-                borderColor: `${theme.main.border}`,
-                margin: "1rem",
-            }}
-            hideFooter
-        />
+                }}
+            >
+                <DialogTitle
+                    sx={{
+                        fontSize: "1.2rem",
+                        textAlign: "center",
+                    }}
+                >
+                    Are you sure you want to submit these rows?
+                </DialogTitle>
+
+                {confirmationErrors.length !== 0 && (
+                    <DialogContent>
+                        <ul>
+                            {Object.entries(countErrorTypes(confirmationErrors)).map(
+                                ([errorType, count]) => (
+                                    <li key={errorType}>
+                                        {errorType}: {count} row(s)
+                                    </li>
+                                )
+                            )}
+                        </ul>
+                    </DialogContent>
+                )}
+                <DialogActions sx={{ justifyContent: "center" }}>
+                    <>
+                        <Button
+                            variant="contained"
+                            sx={{ margin: "1rem", minWidth: "120px" }}
+                            onClick={() => {
+                                setIsConfirmationDialogVisible(false);
+                            }}
+                        >
+                            Close
+                        </Button>
+                        {confirmationErrors.length == 0 && (
+                            <Button
+                                variant="contained"
+                                onClick={handleSubmit}
+                                sx={{ margin: "1rem", minWidth: "120px" }}
+                            >
+                                Submit
+                            </Button>
```

Comment:
Rather than having this conditionally rendered, could we `disable` it based on the errors?